### ISSUE TITLE
Pagina - Prodotto Autovetture Valori numerici

### ISSUE THREAD

**Issue by appfrontIT** [appfrontIT/lookinglass.github.io#3]
PremioNetto: ":823,81↵↵↵↵		"

Il valore del premio viene salvato con caratteri non ammessi.

In generale: lo script dovrebbe riconoscere i valori numerici e salvarli come numero (intero o float) nel json.

**Comment by MeerKatDev** [comment]
Ho corretto e rivisto tutta la procedura, adesso alla fine mette tutto l'oggetto in console e in sessionStorage, un po' per vedere come e' formato e un po' come stub per poi facilmente convertirlo in una chiamata ajax. Puoi verificare se prende tutti i dati correttamente?

**Comment by appfrontIT** [comment]
Ho fatto una verifica, queste le modifiche da fare:

1. Nell'array pagina le pagine vanno inserite nello stesso ordine con cui sono state visitate [prima pagina, seconda pagina, etc.]. Sarebbe utile avere anche una proprietà ID all'interno dell'oggetto pagina (partiamo sempre da 0)
2. Nella pagina "Dati Anagrafici" il form deve essere composto sia da "Dati contraente" che da "Dati proprietario" (non sempre sono gli stessi)
3. Nella pagina "Attestato di Rischio" deve essere riportata la tabella: riga "Sinistri pagati resp. princ." (con i valori di label di ogni colonna), riga "Sinistri pagati resp. parit."
4. Nella pagina "Attestato di Rischio" deve essere riportata la tabella "DETTAGLIO SINISTRI PAGATI CON RESPONSABILITÀ PARITARIA", ricorda che le tabelle sono sempre un array di righe, dove ogni riga è un oggetto che contiene le colonne {label colonna: valore contenuto}
5. Sulla pagina "Riepilogo" i dati numerici della tabella vanno riportati secondo lo schema del punto precedente: le tabelle sono sempre un array di righe, dove ogni riga è un oggetto che contiene le colonne {label colonna: valore contenuto}
6. Se possibile aggiungiamo all'interno dell'oggetto anche l'indirizzo IP della sessione

**Comment by MeerKatDev** [comment]
1. Lo prende in ordine, ma e' un ordine fisso. Siccome per adesso e' un percorso unico, e sono sempre le stesse pagine nello stesso ordine, sarebbe meglio tenerlo cosi, mentre se bisognera' generalizzare per percorsi diversi, adottero' la strategia appropriata.
6. di solito l'IP si prende dal server, nella richiesta http. lato client bisogna fare ineleganti hack per ottenerlo, che consiste comunque nell'ottenerlo da qualche server.

**Comment by appfrontIT** [comment]
Risposte inline

  1.  Lo prende in ordine, ma e' un ordine fisso. Siccome per adesso e' un percorso unico, e sono sempre le stesse pagine nello stesso ordine, sarebbe meglio tenerlo cosi, mentre se bisognera' generalizzare per percorsi diversi, adottero' la strategia appropriata.
>Suggerisco di creare l’oggetto globale all’inizio e fare il push dell’oggetto pagina per ogni pagina caricata all’interno dell’array pagine:[]. In questo modo si mantengono in ordine.

  1.  di solito l'IP si prende dal server, nella richiesta http. lato client bisogna fare ineleganti hack per ottenerlo, che consiste comunque nell'ottenerlo da qualche server.

**Comment by MeerKatDev** [comment]
1. Capisco che ti serve in ordine, e adesso l'array E' in ordine. Non e' possibile usare una variabile globale semplice, perche' il js si resetta ogni volta che la pagina si ricarica. Vorrei evitare una soluzione del tipo tenerlo nello storage, perche' dovrei fare il parsing ogni volta che si carica la pagina. Questa soluzione per adesso e' adatta al tipo di procedimento, quando ci sara' bisogno di espanderla a diversi procedimenti, in quel momento si adatta questo metodo.
2. Capisco che ti serve l'ip, ma puo' essere messo lato server quando mandi il json. Perche' usare un workaround quando si puo' mettere nella richiesta al momento opportuno?

**Comment by MeerKatDev** [comment]
Ho provato a cambiare gli altri punti come richiesto, prova a vedere se adesso va bene. Solo una cosa non comprendo bene, come devo cambiare "Riepilogo". Le righe sono strutturate cosi': `{Imponibile: [855.4, 855.4]}`, devo cambiare questa struttura?

**Comment by appfrontIT** [comment]
Usiamo sempre le risposte inline, per riferimento.

1. Capisco che ti serve in ordine, e adesso l'array E' in ordine. Non e' possibile usare una variabile globale semplice, perche' il js si resetta ogni volta che la pagina si ricarica. Vorrei evitare una soluzione del tipo tenerlo nello storage, perche' dovrei fare il parsing ogni volta che si carica la pagina. Questa soluzione per adesso e' adatta al tipo di procedimento, quando ci sara' bisogno di espanderla a diversi procedimenti, in quel momento si adatta questo metodo.

>>Ok

2. Capisco che ti serve l'ip, ma puo' essere messo lato server quando mandi il json. Perche' usare un workaround quando si puo' mettere nella richiesta al momento opportuno?

>>Giusto

3. Ho provato a cambiare gli altri punti come richiesto, prova a vedere se adesso va bene. Solo una cosa non comprendo bene, come devo cambiare "Riepilogo". Le righe sono strutturate cosi': {Imponibile: [855.4, 855.4]}, devo cambiare questa struttura?

>> Si hai ragione. Non l'ho spiegato bene. Vedi questo esempio sotto

```json
[
  {"riga": "Imponibile",
  "prima rata": 821,
  "seconda rata": 821
  },
  {"riga": "Di cui Diritti",
  "prima rata": 0,
  "seconda rata": 0
  }
]

4. Nella pagina "Prodotto Autovetture" non viene registrato il valore del massimale. 

5. Se nella navigazione torno indietro non vengono aggiornati i dati (es. dall'ultima pagina, sono tornato indietro varie volte e non mi ha aggiornato i nuovi dati).

**Comment by MeerKatDev** [comment]
> 4. Nella pagina "Prodotto Autovetture" non viene registrato il valore del massimale.

Controllo

> 5. Se nella navigazione torno indietro non vengono aggiornati i dati (es. dall'ultima pagina, sono tornato indietro varie volte e non mi ha aggiornato i nuovi dati)

I dati vengono registrati quando clicchi "Prosegui" o il pallino blu quando opportuno.

**Comment by appfrontIT** [comment]
> Va bene cosi'? : {"Imponibile": {"prima rata": 821, "seconda rata": 821 }
Ok

**Comment by MeerKatDev** [comment]
Non riesco ad andare oltre ai Dati Anagrafici con i soliti dati, qualcosa e' cambiato?

**Comment by MeerKatDev** [comment]
Massimale adesso e' inserito, chiudo.
